src/main.zig

```diff
@@ -281,6 +289,50 @@ const CLI = struct {
         try self.printTable(allocator, results);
     }
 
+    fn handleImport(self: @This(), allocator: std.mem.Allocator, input: @TypeOf(@as(Input, undefined).Import)) !void {
+        var bookmarkFile = try paths.getBookmarkFile(allocator, .append);
+        defer bookmarkFile.close();
+        var bookmarkFileBuffer: [1024]u8 = undefined;
+        var writer = bookmarkFile.writer(&bookmarkFileBuffer);
+        var bookmarks: std.ArrayList(Bookmark) = .empty;
+        defer bookmarks.deinit(allocator);
+        var successful_count: usize = 0;
+        var failed_count: usize = 0;
+        defer {
+            for (bookmarks.items) |bookmark| {
+                allocator.free(bookmark.tags);
+            }
+        }
+
+        if (input.path) |path| {
+            // handle importing from file
+            var file = try paths.getFileByPath(allocator, path, .read_only, false);
+            defer file.close();
+            var buffer: [1024]u8 = undefined;
+            var reader = file.reader(&buffer);
+            const parsedBookmarks = try Bookmark.from_csv(allocator, &reader.interface);
+            defer allocator.free(parsedBookmarks.bookmarks);
+            try bookmarks.appendSlice(allocator, parsedBookmarks.bookmarks);
+            successful_count = parsedBookmarks.successful_count;
+            failed_count = parsedBookmarks.failed_count;
+        } else {
+            // handle piping from stdin
+            const parsedBookmarks = try Bookmark.from_csv(allocator, self.stdin);
+            defer allocator.free(parsedBookmarks.bookmarks);
+            try bookmarks.appendSlice(allocator, parsedBookmarks.bookmarks);
+            successful_count = parsedBookmarks.successful_count;
+            failed_count = parsedBookmarks.failed_count;
```

Comment:
No changes necessary here, but just making a note for myself: I think we can accomplish this with streaming between the reader and writer which may be a simpler path forward. This seems fine for now though.